buildSrc/src/main/kotlin/klib/director/KlibDirectorUtils.kt

```diff
@@ -0,0 +1,43 @@
+package klib.director
+
+import java.io.File
+import java.util.zip.ZipFile
+
+internal fun File.filesTxt() = resolve("files.txt")
+
+internal fun File.latestGenerationDir() = resolve("latest")
+
+internal fun File.mainKlib() = resolve("main.klib")
+
+internal fun File.isKlibFile() = path.endsWith(".klib") || path.endsWith(".jar")
+
+internal fun File.moveTo(other: File) {
+    if (!renameTo(other)) {
+        copyTo(other, true)
+        delete()
+    }
+}
+
+internal fun File.deleteInernals(): Boolean = walkBottomUp().fold(true, { res, it ->
+    it == this || ((it.delete() || !it.exists()) && res)
+})
+
+internal fun File.unzipTo(outDir: File) {
```

Comment:
It can be done via `ArchiveOperations.zipTree`. It contains tree, and in task it can be `copied` via `FileSystemOperations`.
```
fs.copy {
    it.from(archiveOperations.zipTree(archive))
    it.into(destination)
}